### PR TITLE
Add blood donation form

### PR DESCRIPTION
**Pull request by soryy708**

### CODE REVIEW

File: templates/medical/blood_donation.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+[center][b]Blood Donation Application Form[/b][/center]
+[hr]
+First time donor? [field] [small](yes/no)[/small][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
Have you donated blood before? [field] [small](yes/no)[/small][br]
```

File: templates/medical/blood_donation.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+[hr]
+[u]For Repeat Donor[/u][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
[u]For Repeat Donors[/u][br]
```

File: templates/medical/blood_donation.txt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Did you encounter any problems in your last donation? [field]
+[small](No problems / fainting / bruise / difficulties finding vein / other)[/small][br]
+[hr]
+Blood group: [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
Blood type: [field][br]
```

Actually when I've given blood they've tested it after sending it to their lab, they wouldn't rely on the donor to know their blood type.

File: templates/medical/blood_donation.txt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+Age: [field][i]year[/i] Sex: [field] Weight: [field][i]kg[/i][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
Date of Birth: [field] Sex: [field] Height: [field] Weight: [field][i]kg[/i][br]
```

Comment (Dibasic) [reply templates/medical/blood_donation.txt]: Date of birth on forms is just a personal preference, if you keep 'age' then I would put 'years' instead of 'year'

File: templates/medical/blood_donation.txt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+If more than 60-70 years old, Do you have medical certificate? [field] [small](yes/no)[/small][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]: Not certain what this line means, may want to rephrase. Also, give a single age, not a range

File: templates/medical/blood_donation.txt [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+Occupation: [field][br]
+Name: [i]Mr. / Ms. / Mrs.[/i] [field] Signature: [field][br]
+[hr]
+[hr]
+[center][b]For your own safety and the safety of the patient who will receive your blood, please answer the following questions, if applicable, to the best of your knowledge:[/b][/center]
+[hr]
+[u]Category 1 [small](For women only)[/small][/u][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
[u]Obstetric History[small](if applicable)[/small][/u][br]
```

File: templates/medical/blood_donation.txt [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+1. Pregnant? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
1. Are you now or do you plan to become pregnant? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+2. Do breast-feed? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
2. Do you breastfeed? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+3. Gave birth or miscarriage in the last 6 months? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
3. Have you given birth or miscarried in the last 6 months? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+[hr]
+[u]Category 2[/u]
+4. Had diarrhea in the last 7 days? [field][br]
+5. Had major surgery in the last 6 months or minor surgery in the last 7 days? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
5. Have you undergone any surgery in the last 6 months? [field][br]
```

I don't know that they'd trust a patient to judge whether or not their surgery was major

File: templates/medical/blood_donation.txt [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+[hr]
+[center][b]For your own safety and the safety of the patient who will receive your blood, please answer the following questions, if applicable, to the best of your knowledge:[/b][/center]
+[hr]
+[u]Category 1 [small](For women only)[/small][/u][br]
+1. Pregnant? [field][br]
+2. Do breast-feed? [field][br]
+3. Gave birth or miscarriage in the last 6 months? [field][br]
+[hr]
+[u]Category 2[/u]
+4. Had diarrhea in the last 7 days? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
4. Have you had diarrhea in the last 7 days? [field][br]
```
suggesting present perfect for all of these

File: templates/medical/blood_donation.txt [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+5. Had major surgery in the last 6 months or minor surgery in the last 7 days? [field][br]
+6. Do you drink alcohol or others? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
6. Do you drink alcohol? [field] If so, how many drinks do you have in a typical week? [field][br]
```
something like this

File: templates/medical/blood_donation.txt [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+7. Had a history of drug use or had you been improsoned in the last 3 years? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
7. Do you have any history of drug addiction or have you been imprisoned in the last 3 years? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+8. Had a blood transfusion in the past 1 year? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
8. Have you had a blood transfusion in the past 12 months? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+[u]Category 3[/u]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt:29]: Is there a specific way this category is different from the others? If so, should label it more specifically

File: templates/medical/blood_donation.txt [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+[hr]
+[hr]
+[center][b]For your own safety and the safety of the patient who will receive your blood, please answer the following questions, if applicable, to the best of your knowledge:[/b][/center]
+[hr]
+[u]Category 1 [small](For women only)[/small][/u][br]
+1. Pregnant? [field][br]
+2. Do breast-feed? [field][br]
+3. Gave birth or miscarriage in the last 6 months? [field][br]
+[hr]
+[u]Category 2[/u]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt:23]: Is there a specific way this category is different from the others? If so, should label it more specifically

File: templates/medical/blood_donation.txt [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+4. Had diarrhea in the last 7 days? [field][br]
+5. Had major surgery in the last 6 months or minor surgery in the last 7 days? [field][br]
+6. Do you drink alcohol or others? [field][br]
+7. Had a history of drug use or had you been improsoned in the last 3 years? [field][br]
+8. Had a blood transfusion in the past 1 year? [field][br]
+[u]Category 3[/u]
+9. Have menstruation? [small](to be answered by female only)[/small] [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
9. Are you menstruating? [small](if applicable)[/small] [field][br]
```

File: templates/medical/blood_donation.txt [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+10. Do you feel fit enough and have enough rest last night? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
10. Do you feel fit and well rested? [field][br]
```
might be awkward

File: templates/medical/blood_donation.txt [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+11. Did you take any medicine / antibiotics? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
11. What medication have you taken in the last 7 days? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+12. Have you or any of your family member ever had hepatitis? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
12. Do you have any history or family history of hepatitis? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+13. Have asthma, epilepsy, chronic skin disease, chronic cough, tuberculosis, allergies, high blood pressure, heart/kidney/thyroid disease, cancer, bleeding disorder, etc? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
13. Do you have asthma, epilepsy, chronic skin disease, chronic cough, tuberculosis, allergies, high blood pressure, heart/kidney/thyroid disease, cancer, any bleeding disorder, or any communicable disease? [small](please list which, if any, apply)[/small] [field][br]
```

File: templates/medical/blood_donation.txt [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+14. Have ear/body piercings, tattoos made or removed or acupuncture? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
14. Have you had any body piercing, any tattoo applied or removed, or any acupuncture in the last 12 months? [field][br]
```
something like this

File: templates/medical/blood_donation.txt [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+15. Did you get any vaccinations in the last 2 months? [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
15. Have you had any vaccinations in the last 2 months? [field][br]
```

File: templates/medical/blood_donation.txt [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+26. Others: [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
16. Other notes or issues: [field][br]
```

File: templates/medical/blood_donation.txt [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+[hr]
+[small]I hereby certify that I have answered the following questions truthfully and that, to the best of my knowledge, my blood is safe for donation.
+I have been informed that my blood will be tested for syphilis, hepatitis B and C, as well as HIV/AIDS.
+I hereby voluntarily donate blood without expecting any type of remuneration.
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
I consent to donate blood without expecting any type of remuneration.
```

File: templates/medical/blood_donation.txt [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+10. Do you feel fit enough and have enough rest last night? [field][br]
+11. Did you take any medicine / antibiotics? [field][br]
+12. Have you or any of your family member ever had hepatitis? [field][br]
+13. Have asthma, epilepsy, chronic skin disease, chronic cough, tuberculosis, allergies, high blood pressure, heart/kidney/thyroid disease, cancer, bleeding disorder, etc? [field][br]
+14. Have ear/body piercings, tattoos made or removed or acupuncture? [field][br]
+15. Did you get any vaccinations in the last 2 months? [field][br]
+26. Others: [field][br]
+[hr]
+[small]I hereby certify that I have answered the following questions truthfully and that, to the best of my knowledge, my blood is safe for donation.
+I have been informed that my blood will be tested for syphilis, hepatitis B and C, as well as HIV/AIDS.
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
I have been informed that my blood will be tested for syphilis, hepatitis B and C, HIV, and other bloodborne illnesses.
```

File: templates/medical/blood_donation.txt [hunk 27]
```diff
@@ -0,0 +1,46 @@
...
+9. Have menstruation? [small](to be answered by female only)[/small] [field][br]
+10. Do you feel fit enough and have enough rest last night? [field][br]
+11. Did you take any medicine / antibiotics? [field][br]
+12. Have you or any of your family member ever had hepatitis? [field][br]
+13. Have asthma, epilepsy, chronic skin disease, chronic cough, tuberculosis, allergies, high blood pressure, heart/kidney/thyroid disease, cancer, bleeding disorder, etc? [field][br]
+14. Have ear/body piercings, tattoos made or removed or acupuncture? [field][br]
+15. Did you get any vaccinations in the last 2 months? [field][br]
+26. Others: [field][br]
+[hr]
+[small]I hereby certify that I have answered the following questions truthfully and that, to the best of my knowledge, my blood is safe for donation.
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
[small]I certify that I have answered the above questions truthfully and that, to the best of my knowledge, my blood is safe for donation.
```

File: templates/medical/blood_donation.txt [hunk 28]
```diff
@@ -0,0 +1,46 @@
...
+I have been informed that my blood will be tested for syphilis, hepatitis B and C, as well as HIV/AIDS.
+I hereby voluntarily donate blood without expecting any type of remuneration.
+The blood may be given to any patient or for research purpose as deemed suitable by Torch medical personnel.
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
I accept that the blood I donate may be given to any patient or used for research purposes as deemed suitable.
```
something like this, also like making this non-Torch-specific

File: templates/medical/blood_donation.txt [hunk 29]
```diff
@@ -0,0 +1,46 @@
...
+I certify that the Torch medical staff are not responsible for any untoward effects that may occour after this blood donation.
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]:
```suggestion
I agree that the staff and technicians involved in my donations are not liable for any untoward effects that may occur after my donation.
```
Just changing the voice of this, also would remove Torch but that's just my opinion

File: templates/medical/blood_donation.txt [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+I shall be pleased to donate blood again.[/small]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]: Not sure what this means

File: templates/medical/blood_donation.txt [hunk 31]
```diff
@@ -0,0 +1,46 @@
...
+Donor signature: [field][br]
+Doctor/Staff signature: [field][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]: Not certain that the nurse involved needs to sign, this form may only need the donor's informed consent

File: templates/medical/blood_donation.txt [hunk 32]
```diff
@@ -0,0 +1,46 @@
+[center][b]Blood Donation Application Form[/b][/center]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]: Would rephrase to "blood donation informed consent form" or something, "application" isn't quite the word for the process imo

**Review by Dibasic — CHANGES_REQUESTED** [review]
Overall this is good and I think a blood donation form is a great addition. There are a lot of comments on this review because of a few repeated suggestions I made for tense and voice.

File: templates/medical/blood_donation.txt (hunk 30 above)
Comment (soryy708) [reply templates/medical/blood_donation.txt]: Took this from a form I found online.

File: templates/medical/blood_donation.txt (hunk 5 above)
Comment (soryy708) [reply templates/medical/blood_donation.txt]: Took this from a form I found online.

File: templates/medical/blood_donation.txt [hunk 33]
```diff
@@ -0,0 +1,46 @@
...
+Blood type: [field][br]
+Date of Birth: [field] Sex: [field] Height: [field] Weight: [field][i]kg[/i][br]
+If more than 60-70 years old, Do you have medical certificate? [field] [small](yes/no)[/small][br]
+Occupation: [field][br]
+Name: [i]Mr. / Ms. / Mrs.[/i] [field] Signature: [field][br]
+[hr]
+[hr]
+[center][b]For your own safety and the safety of the patient who will receive your blood, please answer the following questions, if applicable, to the best of your knowledge:[/b][/center]
+[hr]
+[u]Obstetric History[small](if applicable)[/small][/u][br]
```
Comment (Dibasic) [inline templates/medical/blood_donation.txt]: If we're leaving "Category 2" and "Category 3," may as well label this one "Category 1"
```suggestion
Category 1: [u]Obstetric History[small](if applicable)[/small][/u][br]
```

File: templates/medical/blood_donation.txt (hunk 5 above)
Comment (Dibasic) [reply templates/medical/blood_donation.txt]: I would adjust the age still.
```suggestion
If more than 60 years old, do you have medical certificate? [field] [small](yes/no)[/small][br]
```

File: templates/medical/blood_donation.txt (hunk 30 above)
Comment (Dibasic) [reply templates/medical/blood_donation.txt]: It sounds weird to me, I'd suggest taking out this one line.
```suggestion
[/small]
```

**Review by Dibasic — CHANGES_REQUESTED** [review]
More or less good, just a couple of things to clean up

**Comment by soryy708** [comment]
@Dibasic I think it's ready for review

**Review by Dibasic — APPROVED** [review]
Looks good.
